app/scripts/controller/util.js

```diff
@@ -630,6 +632,158 @@
       return true;
     };
 
+    /**
+     * General pick clinical attributes based on predesigned Regex
+     * This filter is the same one which used in previous Google Charts Version,
+     * should be revised later.
+     * 
+     * @param {string} attr Clinical attribute ID.
+     * @return {boolean} Whether input attribute passed the criteria.
+     */
+    content.isPreSelectedClinicalAttr = function(attr) {
+      return attr.toLowerCase().match(/(os_survival)|(dfs_survival)|(mut_cnt_vs_cna)|(mutated_genes)|(cna_details)|(^age)|(gender)|(sex)|(darwin_vital_status)|(darwin_patient_age)|(os_status)|(os_months)|(dfs_status)|(dfs_months)|(race)|(ethnicity)|(.*type.*)|(.*site.*)|(.*grade.*)|(.*stage.*)|(histology)|(tumor_type)|(subtype)|(tumor_site)|(.*score.*)|(mutation_count)|(copy_number_alterations)|(sequenced)|(has_cna_data)|(sample_count_patient)/);
+    };
+
+    /**
+     * This is the function to define study specific priority clinical
+     * attributes list.
+     * TODO: need to work with priority.
+     * 
+     * @param {string} attr Clinical attribute ID
+     * @param {string} studyId Study ID.
+     * @return {boolean} Whether input attribute is prioritized clinical attribute.
+     */
+    content.isPriorityClinicalAttr = function(attr, studyId) {
+      // Should be all lower case
+      var studySpecific = {
+        mskimpact: []
+      };
+
+      if (_.isString(attr)) {
+        attr = attr.toLowerCase();
+        if (_.isString(studyId) && studySpecific.hasOwnProperty(studyId) &&
+          studySpecific[studyId].indexOf(attr) !== -1) {
+          return true;
+        }
+        if (clinicalAttrsPriority.indexOf(attr) !== -1) {
+          return true;
+        }
+      }
+
+      return false;
+    };
+
+    /**
+     * Compare based on data availability.
+     * Notice that: attribute with only one category will be moved to end.
+     * Number of keys in this attribute is more than half numOfDatum
+     * will be moved to end as well.
+     *
+     * @param {object} a Attribute meta item A.
+     * @param {object} b Attribute meta item B.
+     * @return {number} Indicator which item is selected.
+     */
+    content.compareClinicalAvailability = function(a, b) {
```

Comment:
In the next milestone as we are thinking of lazy loading to clinical data right, then how would we use this function? I mean we won't be having 'numOfDatum' info initially while loading right.